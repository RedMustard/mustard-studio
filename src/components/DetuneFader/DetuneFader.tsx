import { h } from 'preact';
import { MAX_DETUNE, MIN_DETUNE, DETUNE_STEPS } from '../../constants';
import { ValueUnitEnum } from '../../types/runtimeTypes';
import { Fader } from '../Fader/Fader';


interface DetuneFaderProps {
    value: number;
    classSuffix: string;
    onInput: (value: number) => void;
    onMouseDown: (e: MouseEvent) => void;
}

export const DetuneFader = ({
    value,
    onInput,
    onMouseDown,
    classSuffix,
}: DetuneFaderProps) => (
    <Fader
        classSuffix={classSuffix}
        maxValue={MAX_DETUNE}
        minValue={MIN_DETUNE}
        onInput={onInput}
        onMouseDown={onMouseDown}
        stepResolution={DETUNE_STEPS}
        value={value}
        valueUnit={ValueUnitEnum.CENT}
    />
);
