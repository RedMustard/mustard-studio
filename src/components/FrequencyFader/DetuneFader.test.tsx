import { h } from 'preact';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DetuneFader } from './DetuneFader';


const baseProps = {
    value: 1.0,
    onInput: jest.fn(),
    onMouseDown: jest.fn(),
    classSuffix: 'foo',
};

describe('<DetuneFader />', () => {
    it('renders with basic props', () => {
        const wrapper = shallow(
            <DetuneFader
                {...baseProps}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('calls onInput when changed', () => {
        const wrapper = mount(
            <DetuneFader
                {...baseProps}
            />,
        );
        wrapper.find('input').simulate('input');
        expect(baseProps.onInput).toBeCalled();
    });

    it('calls onMouseDown when clicked', () => {
        const wrapper = mount(
            <DetuneFader
                {...baseProps}
            />,
        );
        wrapper.find('input').simulate('mousedown');
        expect(baseProps.onInput).toBeCalled();
    });
});
