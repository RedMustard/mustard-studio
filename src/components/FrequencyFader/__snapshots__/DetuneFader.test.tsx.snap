// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetuneFader /> renders with basic props 1`] = `
<Fader
  classSuffix="foo"
  maxValue={1200}
  minValue={-1200}
  onInput={[MockFunction]}
  onMouseDown={[MockFunction]}
  stepResolution={100}
  value={1}
  valueUnit="cent"
/>
`;
